cpp/src/arrow/flight/sql/CMakeLists.txt

```diff
@@ -117,10 +117,4 @@ if(ARROW_BUILD_TESTS OR ARROW_BUILD_EXAMPLES)
   add_executable(flight-sql-test-app test_app_cli.cc)
   target_link_libraries(flight-sql-test-app PRIVATE ${ARROW_FLIGHT_SQL_TEST_LINK_LIBS}
                                                     ${GFLAGS_LIBRARIES})
-
-  if(ARROW_FLIGHT_TEST_LINKAGE STREQUAL "static" AND ARROW_BUILD_STATIC)
-    target_compile_definitions(flight_sql_test
```

Comment:
Oops, my bad. We still need this block, though, I believe. Just replace `flight_sql_test` with `arrow-flight-sql-test`.

Reply:
it does fail on that case:
```
CMake Error at src/arrow/flight/sql/CMakeLists.txt:122 (target_compile_definitions):
  target_compile_definitions called with invalid arguments
```
with:
```
  if(ARROW_FLIGHT_TEST_LINKAGE STREQUAL "static" AND ARROW_BUILD_STATIC)
    target_compile_definitions(arrow-flight-sql-test
                               flight-sql-test-server flight-sql-test-app
                               PUBLIC ARROW_FLIGHT_STATIC ARROW_FLIGHT_SQL_STATIC)
  endif()
```

Reply:
```diff
   if(ARROW_FLIGHT_TEST_LINKAGE STREQUAL "static" AND ARROW_BUILD_STATIC)
-    target_compile_definitions(flight_sql_test
-                               flight-sql-test-server flight-sql-test-app
-                               PUBLIC ARROW_FLIGHT_STATIC ARROW_FLIGHT_SQL_STATIC)
+    foreach(TEST_TARGET arrow-flight-sql-test flight-sql-test-server flight-sql-test-app)
+      target_compile_definitions(${TEST_TARGET}
+                                PUBLIC ARROW_FLIGHT_STATIC ARROW_FLIGHT_SQL_STATIC)
+    endforeach()
   endif()
```